src/main/java/net/dv8tion/jda/api/entities/MessageChannel.java

```diff
@@ -1565,7 +1565,68 @@ default MessagePaginationAction getIterableHistory()
         return getHistoryBefore(message.getId(), limit);
     }
-
+    
+    /**
+     * Retrieves messages from the beginning of this MessageChannel.
+     * The {@code limit} determines the amount of messages being retrieved.
+     * 
+     * <p><b>Example:</b>
+     * <pre><code>
+     * public void resendFirstMessage(MessageChannel channel)
+     * {
+     *     channel.getHistoryFromBeginning(1).queue(history ->
+     *     {
+     *         if (!history.isEmpty())
+     *         {
+     *             Message firstMsg = history.getRetrievedHistory().get(0);
+     *             channel.sendMessage(firstMsg).queue();
+     *         }
+     *         else
+     *             channel.sendMessage("No history for this channel!").queue();
+     *     });
+     * }
+     * </code></pre>
+     *
+     * <p>The following {@link net.dv8tion.jda.api.requests.ErrorResponse ErrorResponses} are possible:
+     * <ul>
+     *     <li>{@link net.dv8tion.jda.api.requests.ErrorResponse#MISSING_ACCESS MISSING_ACCESS}
+     *     <br>The request was attempted after the account lost access to the {@link net.dv8tion.jda.api.entities.Guild Guild}
+     *         typically due to being kicked or removed, or after {@link net.dv8tion.jda.api.Permission#MESSAGE_READ Permission.MESSAGE_READ}
+     *         was revoked in the {@link net.dv8tion.jda.api.entities.TextChannel TextChannel}</li>
+     *
+     *     <li>{@link net.dv8tion.jda.api.requests.ErrorResponse#MISSING_PERMISSIONS MISSING_PERMISSIONS}
+     *     <br>The request was attempted after the account lost
+     *         {@link net.dv8tion.jda.api.Permission#MESSAGE_HISTORY Permission.MESSAGE_HISTORY} in the
+     *         {@link net.dv8tion.jda.api.entities.TextChannel TextChannel}.</li>
+     *
+     *     <li>{@link net.dv8tion.jda.api.requests.ErrorResponse#UNKNOWN_CHANNEL UNKNOWN_CHANNEL}
+     *     <br>The request was attempted after the channel was deleted.</li>
+     * </ul>
+     *
+     * @param  limit
+     *         The amount of messages to be retrieved. Minimum: 1, Max: 100.
+     *
+     * @throws java.lang.IllegalArgumentException
+     *         Provided {@code limit} is less than {@code 1} or greater than {@code 100}.
+     * @throws net.dv8tion.jda.api.exceptions.InsufficientPermissionException
+     *         If this is a {@link net.dv8tion.jda.api.entities.TextChannel TextChannel} and the logged in account does not have
+     *         <ul>
+     *             <li>{@link net.dv8tion.jda.api.Permission#MESSAGE_READ Permission.MESSAGE_READ}</li>
+     *             <li>{@link net.dv8tion.jda.api.Permission#MESSAGE_HISTORY Permission.MESSAGE_HISTORY}</li>
+     *         </ul>
+     *
+     * @return {@link net.dv8tion.jda.api.entities.MessageHistory.MessageRetrieveAction MessageHistory.MessageRetrieveAction}
+     *         <br>Provides a {@link net.dv8tion.jda.api.entities.MessageHistory MessageHistory} object with messages - beginning from the start - loaded into it.
```

Comment:
"beginning from the start" sounds redundant to me

Reply:
Yes, true. But all the other methods also explain in the return tag which kind of Messages will be in the history and I tried to include that the first messages will be in it.

```suggestion
     *         <br>Provides a {@link net.dv8tion.jda.api.entities.MessageHistory MessageHistory} object with the first messages of the channel loaded into it.
```
Is that better?

Reply:
But the phrase "beginning from the start" doesn't really clarify anything. Its simply redundant and confusing to read, imo

Reply:
My problem is that `Provides a MessageHistory object with messages loaded into it.` says nothing at all. Without the method name I would not know which messages I'll get.

My suggestion doesn't have this phrase in it anymore, is that better? If not, can you suggest me something because I don't know something better.

Reply:
I would just say "the oldest messages in this channel" but first would be fine as well. Maybe add another sentence: "The messages are ordered from the most recent to oldest" for clarity

Reply:
> Maybe add another sentence: "The messages are ordered from the most recent to oldest" for clarity

Isn't it the other way round? The first element is the oldest message for sure, I tested that.

Reply:
Nope, it definitely should be ordered most recent to oldest. I also tested this.

Reply:
You're totally right. Back then when testing it, I just retrieved one message, that's why I thought `get(0)` returns the oldest message.. :man_facepalming: